src/Model/Widget/SpanParser.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpTui\Tui\Model\Widget;
+
+use PhpTui\Tui\Model\Color;
+use PhpTui\Tui\Model\Color\AnsiColor;
+use PhpTui\Tui\Model\Color\RgbColor;
+use PhpTui\Tui\Model\Modifier;
+use PhpTui\Tui\Model\Style;
+
+final class SpanParser
+{
+    private const OPEN_TAG_REGEX = '[a-z](?:[^\\\\<>]*+ | \\\\.)*';
+    private const CLOSE_TAG_REGEX = '[a-z][^<>]*+';
+
+    /** @var array<Style> */
+    private array $styleStack = [];
```

Comment:
maybe avoid state and pass `&$styleStack` in the function calls?

Reply:
Sounds reasonable. I'll do it.